src/server/scripts/Northrend/Naxxramas/boss_gluth.cpp

```diff
@@ -110,47 +133,333 @@ class boss_gluth : public CreatureScript
+    spell_gluth_zombiechow_search() : SpellScriptLoader("spell_gluth_zombiechow_search") { }
+
+    class spell_gluth_zombiechow_search_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_gluth_zombiechow_search_SpellScript);
+
+        void HealForEachTargetHit()
+        {
+            GetOriginalCaster()->ModifyHealth(int32(GetOriginalCaster()->CountPctFromMaxHealth(5)));
+        }
+        
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (sSpellMgr->GetSpellInfo(SPELL_ZOMBIE_CHOW_SEARCH_MULTI) && sSpellMgr->GetSpellInfo(SPELL_ZOMBIE_CHOW_SEARCH_SINGLE))
+                return true;
             else
+                return false;
+        }
+
+        void Register() override
+        {
+            AfterHit += SpellHitFn(spell_gluth_zombiechow_search_SpellScript::HealForEachTargetHit);
+        }
+    };
+
+    SpellScript* GetSpellScript() const
+    {
+        return new spell_gluth_zombiechow_search_SpellScript();
+    }
+
+};
+
+// creature 16360 (10man) / 30303 (25man)
+class npc_zombie_chow : public CreatureScript
+{
+public:
+
+    npc_zombie_chow() : CreatureScript("npc_zombie_chow") { }
+
+    struct npc_zombie_chowAI : public ScriptedAI
+    {
+        npc_zombie_chowAI(Creature* creature) : ScriptedAI(creature)
+        {
+            GluthGUID = creature->GetInstanceScript()->GetGuidData(DATA_GLUTH);
+
+            DoCast(me, SPELL_INFECTED_WOUND);
+            state = STATE_ZOMBIE_NORMAL;
+        }
+
+        void UpdateAI(uint32 /*diff*/) override
+        {
+            if (!UpdateVictim())
+                return;
+
+            Creature* gluth = ObjectAccessor::GetCreature(*me, GluthGUID);
+            if (state == STATE_ZOMBIE_DECIMATED)
+            {
+                // Putting this in the UpdateAI loop fixes an issue where death gripping a decimated zombie would make the zombie stand still until the rest of the fight.
+                // Also fix the issue where if one or more zombie is rooted when decimates hits (and MovePoint() is called), the zombie teleport to the boss. pretty weird behavior.
+                if (me->IsStopped() && me->CanFreeMove() && me->GetExactDist2d(gluth) > 10.0)
+                {
+                    if (Creature* gluth = ObjectAccessor::GetCreature(*me, GluthGUID))
+                        me->GetMotionMaster()->MovePoint(0, gluth->GetPosition()); // isnt dynamic. point to point move without taking into account that Gluth might change his position before reaching destination point.
+                        //me->GetMotionMaster()->MoveFollow(gluth, 0, 0); // the zombies will stand still too far away from Gluth at the end of their movement. Also doesnt seem to work with SetWalk(true)
+                        //me->GetMotionMaster()->MoveChase(gluth); // crash server. I assume its because Gluth is friendly. We need the zombies to get really close to Gluth's center position anyway (10 yards) for the multi kill spell to work.
+
```

Comment:
try to use an event every 1 or 2 seconds with 

````c++
me->GetMotionMaster()->MovePoint(0, gluth->GetPosition());
````

that'll do the trick i think